### DISCUSSION TITLE
GPUI WASM

### DISCUSSION THREAD

**Discussion by fraidev** [zed-industries/zed#8203] [Feature Requests]
Does the GPUI support WASM?
If not, is it on the roadmap?

**Comment by tpisto** [comment]
I have related question to this. Is Zed thinking to use "wgpu" package for platform agnostic GPU rendering? That would also then support WASM.

**Comment by philcockfield** [comment]
Thanks @fraidev and @tpisto for asking this.
I came here for this too.  If Zed can become an in-browser composable component, that would be amazing (and an alternative to Monaco)

Is there any universe in which this would be feasible (RUST → GPUI/WASM → Browser)?

**Comment by bajrangCoder** [comment]
Currently there is no support for wasm but Zed team have plan for it. At this time , there is no work on wasm support.

I think zed will use Blade for graphical stuffs.

Issue: https://github.com/zed-industries/zed/issues/5396

**Comment by JMLX42** [comment]
Hello there!

Any update on the WASM support roadmap? Because WASM support + [gpui-component](https://github.com/longbridge/gpui-component) would make it a giant leap forward for x-platform Rust UI.

**Comment by gm0stache** [comment]
"Zed on the web" is a milestone listed on [the roadmap](https://zed.dev/roadmap).

i think WASM support for GPUI would fall under it. 

note its listed in the section "beyond Zed 1.0", so guess its not something to expect "soon". 

should we open something like a gofundme for someone to work on this - like now?
comment with a 🚀 if you would throw some money at this.

i would simply suggest to require the person working on this to setup github donations.
then we could pay the money as soon, as the stuff is merged + some added period to make sure it fundamentally works.
as proof that it does, I would like to see a "desktop app doing regular stuff" work with it. all of Zed could be too difficult.

**Reply by JMLX42** [reply]
> i think WASM support for GPUI would fall under it.

GPUI in WASM is - IMHO - a lot simpler to implement than the whole of "Zed on the Web". Especially because it fits right with [wasi:gfx](https://github.com/WebAssembly/wasi-gfx)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by jlucaso1** [comment]
Maybe this can help to achieve web support? https://github.com/zed-industries/zed/pull/46758
